packages/core/src/semantic-icon-provider/SemanticIconProvider.tsx

```diff
@@ -0,0 +1,84 @@
+import {
+  CalendarIcon,
+  ChevronDownIcon,
+  ChevronLeftIcon,
+  ChevronRightIcon,
+  ChevronUpIcon,
+  CloseIcon,
+  ErrorSolidIcon,
+  InfoSolidIcon,
+  OverflowMenuIcon,
+  SuccessTickIcon,
+  TearOutIcon,
+  UserSolidIcon,
+  WarningSolidIcon,
+} from "@salt-ds/icons";
+import {
+  type ElementType,
+  type ReactNode,
+  createContext,
+  useContext,
+} from "react";
+
+export type SemanticIconMap = {
+  ChevronDownIcon: ElementType;
+  ChevronDownFilledIcon: ElementType;
+  ChevronUpIcon: ElementType;
+  ChevronUpFilledIcon: ElementType;
+  ErrorSolidIcon: ElementType;
+  SuccessTickIcon: ElementType;
+  InfoSolidIcon: ElementType;
+  WarningSolidIcon: ElementType;
+  ChevronRightIcon: ElementType;
+  OverflowMenuIcon: ElementType;
+  UserSolidIcon: ElementType;
+  CalendarIcon: ElementType;
+  ChevronLeftIcon: ElementType;
+  CloseIcon: ElementType;
+  ChevronRightFilledIcon: ElementType;
+  TearOutIcon: ElementType;
+};
+
+export interface SemanticIconProviderProps {
+  /**
+   * Custom mapping of icon names to components. Overrides default icons if provided.
+   */
+  iconMap?: Partial<SemanticIconMap>;
+
+  /**
+   * Child elements that will use the provided icons.
+   */
+  children: ReactNode;
+}
+
+const defaultIconMap: SemanticIconMap = {
+  ChevronDownFilledIcon: ChevronDownIcon,
+  ChevronUpFilledIcon: ChevronUpIcon,
+  ChevronRightFilledIcon: ChevronRightIcon,
+  ChevronUpIcon,
+  ChevronDownIcon,
+  ErrorSolidIcon,
+  SuccessTickIcon,
+  InfoSolidIcon,
+  WarningSolidIcon,
+  ChevronRightIcon,
+  ChevronLeftIcon,
+  OverflowMenuIcon,
+  UserSolidIcon,
+  CalendarIcon,
+  CloseIcon,
+  TearOutIcon,
+};
+
+const SemanticIconContext = createContext<SemanticIconMap>(defaultIconMap);
+
+export const SemanticIconProvider = ({
+  iconMap = {},
+  children,
+}: SemanticIconProviderProps) => (
+  <SemanticIconContext.Provider value={{ ...defaultIconMap, ...iconMap }}>
```

Comment:
This way of default will not work if provider is not used?

Can you add tests for different scenarios?

Reply:
default will work, as we initialised the context with defaultIconMap
also added tests

Reply:
Most user won't use this provider, therefore no context?

Reply:
added default value as a fallback if there would be no context